services/app-web/src/common-code/healthPlanFormDataType/healthPlanFormData.ts

```diff
@@ -180,17 +183,22 @@ function packageName(
 }
 
 const generateRateName = (
-    rateData: RateDataType,
-    submissionName: string
+    rateData: HealthPlanFormDataType,
+    statePrograms: ProgramArgType[]
 ): string => {
     const {
         rateType,
         rateAmendmentInfo,
         rateDateCertified,
         rateDateEnd,
         rateDateStart,
+        rateProgramIDs,
     } = rateData
-    let rateName = `${submissionName}-RATE`
+    let rateName = `${packageName(
+        rateData,
+        statePrograms,
+        rateProgramIDs
+    )}-RATE`
```

Comment:
I like this change- its more self contained and more trustworthy to unit test rather than passing in the already formatted `submissionName` as an opaque string